include/irr/builtin/shaders/loaders/gltf/fragment_impl.glsl

```diff
@@ -0,0 +1,42 @@
+#include "irr/builtin/shaders/loaders/gltf/common.glsl"
+
+#ifndef _IRR_FRAG_INPUTS_DEFINED_
+#define _IRR_FRAG_INPUTS_DEFINED_
+layout (location = 0) in vec3 LocalPos;
+layout(location = 1) in vec3 Normal;
+layout (location = 2) in vec3 ViewPos;
+#ifndef _DISABLE_UV_ATTRIBUTES
+layout (location = _IRR_UV_ATTRIBUTE_BEGINING_ID) in vec2 UV_0;
+#endif
+#ifndef _DISABLE_COLOR_ATTRIBUTES
+layout (location = _IRR_COLOR_ATTRIBUTE_BEGINING_ID) in vec4 Color_0;
+#endif // _DISABLE_COLOR_ATTRIBUTES
+#endif //_IRR_FRAG_INPUTS_DEFINED_
+
+#ifndef _IRR_FRAG_OUTPUTS_DEFINED_
+#define _IRR_FRAG_OUTPUTS_DEFINED_
+layout (location = 0) out vec4 OutColor;
+#endif //_IRR_FRAG_OUTPUTS_DEFINED_
+
+#if !defined(_IRR_FRAG_SET3_BINDINGS_DEFINED_) && !defined(_DISABLE_UV_ATTRIBUTES)
+#define _IRR_FRAG_SET3_BINDINGS_DEFINED_
+layout (set = 3, binding = 0) uniform sampler2D texture_0;
+#endif //_IRR_FRAG_SET3_BINDINGS_DEFINED_
+
```

Comment:
ok, now here like in the MTL builtin shader, you need to declare `bsdf_cos_eval` and `lighting` functions

as a bonus with @Crisspl 's help you could implement`bsdf_cos_generate` and `bsdf_cos_remainder_and_pdf` functions, then you could do sexy cubemap IBL renders in your example.